src/compute/src/render/reduce.rs

```diff
@@ -510,15 +513,17 @@ where
 
     // If `distinct` is set, we restrict ourselves to the distinct `(key, val)`.
     if distinct {
-        // TODO(#16549): Use explicit arrangement
-        partial = partial.distinct_core();
+        partial = partial
+            .arrange_by_self_named("Arranged ReduceInaccumulable")
+            .threshold_named(debug_name, |_, _| 1)
```

Comment:
I didn't actually run this, but it compiles :)

Reply:
Oops, thanks.

Reply:
So my naive rewrite of this was

```rust
        partial = partial
            .arrange_named::<RowSpine<_, _, _, _>>("Arranged ReduceInaccumulable")
            .reduce_abelian::<_, RowSpine<_, _, _, _>>(
                "ReduceInaccumulable",
                move |_: &(Row, Row), _, t| t.push(((), 1)),
            )
            .as_collection(|k: &(Row, Row), _| k.clone())
```

It seems likely to me that your version is preferable, but after reading reduce_abelian, consolidate_slice, and friends a bit, I think it will be faster if could you elaborate on why?

Also, how do we feel about introducing `format!` string consing here?  I considered introducing `debug_name` into the names where possible, but wasn't sure how we felt about probably-unnecessary allocations, plus the nodes in the memory visualizer were already getting pretty wide with my long names as it is, but someone who uses that interface more might have better input here.